src/OpenIddict.Core/OpenIddictController.cs

```diff
@@ -13,31 +13,13 @@
 using AspNet.Security.OpenIdConnect.Server;
 using Microsoft.AspNet.Authorization;
 using Microsoft.AspNet.Http.Authentication;
-using Microsoft.AspNet.Identity;
-using Microsoft.AspNet.Identity.EntityFramework;
 using Microsoft.AspNet.Mvc;
-using Microsoft.Data.Entity;
 using Microsoft.Extensions.DependencyInjection;
-using Microsoft.Extensions.Internal;
 using Microsoft.IdentityModel.Protocols.OpenIdConnect;
 
 namespace OpenIddict {
-    // Note: this controller is generic and doesn't need to
-    // be marked as internal to prevent MVC from discovering it.
-    public class OpenIddictController<TContext, TUser, TRole, TKey> : Controller
-        where TContext : OpenIddictContext<TUser, TRole, TKey>
-        where TUser : IdentityUser<TKey>
-        where TRole : IdentityRole<TKey>
-        where TKey : IEquatable<TKey> {
-        public OpenIddictController([NotNull] TContext context) {
-            Context = context;
-        }
-
-        /// <summary>
-        /// Gets the OpenIddict context.
-        /// </summary>
-        protected TContext Context { get; }
-
+    // Note: this controller is generic and doesn't need to be marked as internal to prevent MVC from discovering it.
+    public class OpenIddictController<TUser, TApplication> : Controller where TUser : class where TApplication : class {
```

Comment:
Sure we can use constructor injection here, but we still need the two generic parameters, since we need them for `OpenIddict<TApp, TUser>` and `UserManager<TUser>`.